.github/workflows/snl-ci.yml

```
@@ -88,3 +88,78 @@ jobs:
       - name: test_kokkos
         working-directory: kokkos/build
         run: ctest --output-on-failure --timeout 3600
+
+  INTEL_2024_2_1_SPR:
+    name: SNL_OPENMP_SERIAL_INTEL_2024_2_1
+    runs-on: [inteloneapi-basekit-2024.2.1-0-devel-ubuntu22.04-latest-spr]
+
+    steps:
+      - name: checkout_kokkos
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          repository: kokkos/kokkos
+          path: kokkos
+
+      - name: configure_kokkos
+        run: |
+          cd kokkos
+          cmake -S . -B build \
+            -DCMAKE_CXX_COMPILER=icpx \
+            -DCMAKE_CXX_STANDARD=20 \
+            -DCMAKE_CXX_FLAGS='-Werror -fp-model=precise -Wno-pass-failed' \
+            -DKokkos_ENABLE_OPENMP=ON \
+            -DKokkos_ENABLE_SERIAL=ON \
+            -DKokkos_ARCH_SPR=ON \
+            -DCMAKE_CXX_EXTENSIONS=OFF \
+            -DKokkos_ENABLE_DEPRECATED_CODE_4=OFF \
+            -DKokkos_ENABLE_TESTS=ON \
+            -DKokkos_ENABLE_EXAMPLES=ON \
+            -DKokkos_ENABLE_BENCHMARKS=ON
+
+      - name: build_and_install_kokkos
+        working-directory: kokkos
+        run: |
+          cmake --build build --parallel $(nproc)
+          cmake --install build --prefix install
+
+      - name: test_kokkos
+        working-directory: kokkos/build
+        run: ctest --output-on-failure --timeout 3600
+
+  INTEL_2024_2_1_PVC:
+    name: SNL_SYCL_INTEL_2024_2_1
+    runs-on: [inteloneapi-basekit-2024.2.1-0-devel-ubuntu22.04-latest-pv]
+
+    steps:
+      - name: checkout_kokkos
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          repository: kokkos/kokkos
+          path: kokkos
+
+      - name: configure_kokkos
+        run: |
+          cd kokkos
+          cmake -S . -B build \
+            -DCMAKE_CXX_COMPILER=icpx \
+            -DCMAKE_CXX_STANDARD=20 \
+            -DCMAKE_BUILD_TYPE=Release \
+            -DCMAKE_CXX_FLAGS='-Werror -fp-model=precise -fsycl-device-code-split=per_kernel -Wno-pass-failed' \
```

Is `-Wno-pass-failed` necessary if we don't enable `OpenMP`?
```suggestion
            -DCMAKE_CXX_FLAGS='-Werror -fp-model=precise -fsycl-device-code-split=per_kernel \
```

---

@masterleinad good question I'll check, with the OpenMP build I added the flag to avoid -Werror of type
```
# error: loop not vectorized: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Werror,-Wpass-failed=transform-warning]
```

---

@masterleinad you were right, that was only needed for OpenMP 👍